x-pack/plugins/reporting/public/management/mount_management_section.tsx

```diff
@@ -5,55 +5,54 @@
  * 2.0.
  */
 
-import { I18nProvider } from '@kbn/i18n-react';
-import * as React from 'react';
+import React, { Suspense, lazy } from 'react';
 import { render, unmountComponentAtNode } from 'react-dom';
 import { Observable } from 'rxjs';
-import { CoreSetup, CoreStart } from '@kbn/core/public';
-import { ILicense } from '@kbn/licensing-plugin/public';
-import { KibanaThemeProvider } from '@kbn/react-kibana-context-theme';
-import { ReportingAPIClient, InternalApiClientProvider } from '../lib/reporting_api_client';
-import { IlmPolicyStatusContextProvider } from '../lib/ilm_policy_status_context';
-import { ClientConfigType } from '../plugin';
+
+import { EuiLoadingSpinner } from '@elastic/eui';
+
+import type { CoreSetup, CoreStart } from '@kbn/core/public';
+import type { ILicense } from '@kbn/licensing-plugin/public';
+import { ErrorBoundary, ErrorBoundaryKibanaProvider } from '@kbn/shared-ux-error-boundary';
+
+import type { ReportingAPIClient } from '../lib/reporting_api_client';
+import type { ClientConfigType } from '../plugin';
 import type { ManagementAppMountParams, SharePluginSetup } from '../shared_imports';
-import { KibanaContextProvider } from '../shared_imports';
-import { ReportListing } from '.';
-
-export async function mountManagementSection(
-  coreSetup: CoreSetup,
-  coreStart: CoreStart,
-  license$: Observable<ILicense>,
-  config: ClientConfigType,
-  apiClient: ReportingAPIClient,
-  urlService: SharePluginSetup['url'],
-  params: ManagementAppMountParams
-) {
+
+const ManagementApp = lazy(() =>
+  // component is not the default export of the module
+  import('./management_app').then((module) => ({ default: module.ManagementApp }))
+);
+
+interface MountSectionOpts {
+  coreSetup: CoreSetup;
+  coreStart: CoreStart;
+  license$: Observable<ILicense>;
+  config: ClientConfigType;
+  apiClient: ReportingAPIClient;
+  urlService: SharePluginSetup['url'];
+  params: ManagementAppMountParams;
+}
+
+export function mountManagementSection(opts: MountSectionOpts) {
+  const { coreSetup, coreStart, license$, config, apiClient, urlService, params } = opts;
+
   render(
-    <KibanaThemeProvider theme={{ theme$: params.theme$ }}>
-      <I18nProvider>
-        <KibanaContextProvider
-          services={{
-            http: coreSetup.http,
-            application: coreStart.application,
-            uiSettings: coreStart.uiSettings,
-            docLinks: coreStart.docLinks,
-          }}
-        >
-          <InternalApiClientProvider apiClient={apiClient}>
-            <IlmPolicyStatusContextProvider>
-              <ReportListing
```

Comment:
@vadimkibana The ReportListing was lazy loaded in the original code with the line: `import('./management/mount_management_section')` in plugin.ts, but was done in a way that couldn't be wrapped in the new Error Boundary. This PR changes _how_ the component is lazy loaded, so that errors can be caught.

If I were to use your suggestion and change nothing else in the Reporting plugin, then if `plugin.ts` fails when trying to import the chunk of code for the management section, nothing would be caught - the code that we failed to load is what includes the Error Boundary.

My changes were to make an Error Boundary surround content that is expected to be lazy loaded. This arrangement also works if the chunk is loaded, but throws some other error in the rendering.

Reply:
> If I were to use your suggestion and change nothing else in the Reporting plugin, then if `plugin.ts` fails when trying to import the chunk of code for the management section, nothing would be caught since the code that failed to load is what includes the Error Boundary.

That is fine, we are not aiming to catch all possible errors. If you wrap it around `<ReportingListing>` it will catch everything inside of it. The dynamic import error, which is used in the mount function, will be addressed later.

Reply:
Resolved in 4700a450adf and bda5264ef29